lib/engine-native/src/artifact.rs

```diff
@@ -545,13 +555,25 @@ impl NativeArtifact {
         })?;
         let metadata_slice: &'static [u8] =
             slice::from_raw_parts(&size[10] as *const u8, metadata_len as usize);
-        let metadata: ModuleMetadata = bincode::deserialize(metadata_slice)
+
```

Comment:
Since endianness can vary between platforms, can we add an extra check on the `triple.endianness()` assuring that the target triple endianness is the same as host endianness, and returning an error otherwise?

This will shield us from `rkyv` serialization issues in the future when the data is encoded in one endianness in the host but expected as other endianness in when deserialized in the target.

Reply:
Good point!